### DISCUSSION TITLE
Is there a way to make Bots use Vehicles on not supported maps?

### DISCUSSION THREAD

**Discussion by revnu** [Joe91/fun-bots#293] [General]
Hi,

I would love to make Bots use Vehicles on some of the maps that are not supported yet. Someone knows where to start if I want to port this from the maps that already work?

rev

**Comment by koopa990** [comment]
Are you familiar with making AI paths?

**Comment by revnu** [comment]
Not yet - but if this is a working way to make bots use vehicles I would dig into that....

**Comment by koopa990** [comment]
If you really want to make your own stuff, sure.

But, nonetheless to say the least-- Its my main duty to make new nodes, and remake nodes to further improve gameplay flow. IF I can ask you to ONLY modify vehicle data, please do so.

**Reply by revnu** [reply]
Nah, I only would do "my own stuff", if Joe is to busy implementing this. Last time I was modding was BF2 10 years before (LUA) :-)

But I think this is something many players would like to see. So I would also be happy and just wait if it comes on the roadmap. So there will be a day there it just will work on all maps. Also I would help accomplishing that, if someone gives me little guidance.

**Reply by koopa990** [reply]
if thats how you want things to be, then i wont stop you.

Safe to say I am the main admin of making nodes. In accordance to your presence, I have a few things to tell you-- if you are willing to aid me in my work. specifically, the skills to make the most out of bots navigating the map geometry area

**Reply by revnu** [reply]
Once I saw Jets on Kharg Island that just fly a straight line if there where no enemy vehicles in air. So from a not-professional point of view it looked to me, that they just follow a start-vector and then cruise in different areas, switching to attack if other jets come withing range. So you have to define those areas, program them, but the logic then bots have a dogfight can be ported, is it that way?

**Reply by revnu** [reply]
I would look into the mod-files of the maps that are already working with vehicles, and test porting it to one map. If the result is working I would forward it to you and work me tru the other maps.

**Reply by koopa990** [reply]
they only fly on the existing path, no advanced AI smart.

**Reply by revnu** [reply]
I would say I experiment with one map for the beginning, so see if my skill is enough to get usable results. 

I looked in the .map from wake-island. These lines have something to do how it works right?

`53;4;-3.438056;101.223373;-220.263351;3;{"Action":{"yaw":0.31528490781784,"type":"vehicle","pitch":-0.13705691695213,"inputs":[35],"time":0.5}}`

Looks like much much trial&error then I dont have the right tools to assist me :-)

For the beginning I would try to make them fly/drive a straight line. Where is the command to initiate, that the bots spawn into the vehicle?

**Reply by KrazyIvan777** [reply]
as far as i know, anyone is welcomed to create new traces for vehicles and infantry on unsupported maps. Here is a link to a few tutorials on how to create traces, if you want to read up on that first. These are the same tutorials i used when i first started creating traces for vehicles. I would be more than glad to help you if you want to learn how to use the node editor. give me a message on the discord server if you have any questions !
https://github.com/Joe91/fun-bots/wiki/Traces

Fun-Bots comes with a trace editor in which you can create or "record" traces by yourself in-game. If you create some good traces that play well, Joe would be able to add them into the next update ! Happy modding :)
